pkg/apis/kops/validation/legacy.go

```diff
@@ -731,3 +671,43 @@ func DeepValidate(c *kops.Cluster, groups []*kops.InstanceGroup, strict bool) er
 
 	return nil
 }
+
+func validateKubelet(k *kops.KubeletConfigSpec, c *kops.Cluster, kubeletPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if k != nil {
+
+		{
+			// Flag removed in 1.6
+			if k.APIServers != "" {
+				allErrs = append(allErrs, field.Forbidden(
+					kubeletPath.Child("apiServers"),
+					"api-servers flag was removed in 1.6"))
+			}
+		}
+
+		if c.IsKubernetesGTE("1.10") {
```

Comment:
This will panic if the version in the cluster spec is unparseable. Not something that should happen in the validation code. It should keep the old logic of checking `kubernetesRelease`.

Reply:
Good catch. I think that getting this far with an unparseable kubernetes version is undefined behaviour though. While I like outputting all errors (or as many as possible) in one go, I think an unparseable kubernetes version should just return the error immediately.

Reply:
I wouldn't agree it counts as "undefined behavior". It is, admittedly, a hard thing to unit test for. I could see us deciding that an unparseable Kubernetes version is one error we don't try to combine with others.

Reply:
Given how much validation logic depends on a parseable Kubernetes version, I would be in favor of returning a single error early if it isn't parseable.

Reply:
If I read the code correctly, the old code just continued on with 1.15 as the kubernetes version. Changed this to returning early now.

Reply:
I suppose another approach would be to DeepCopy the Cluster and put 1.15 in the copy's KubernetesVersion.